### ISSUE TITLE
Some bugs  about   bmp  session  operate with  bgp-ls  address  family  

### ISSUE THREAD

**Issue by tankII** [osrg/gobgp#2686]
When sending bmp  encapsulated  BGP-LS update messages with the bmp client in gobgpd, the following issues were observed:

1. After the bmp station (openbmpd) restarts, the bmp client in gobgpd does not attempt to rebuild the connection or retransmit the bgp-ls messages; but  when test with ipv4 unicast address family  the connection is successfully reestablished  and messages are sent normally .

2. In a scenario where the BMP session has already been established between the bmp station and gobgpd, if gobgp first deletes the bmp session and then adds it again, the BGP-LS messages encapsulated and sent by bmp have incorrect formatting. As a result, openbmpd is unable to parse them correctly. Analyzing the captured bmp messages using Wireshark reveals that the bmp header can be parsed, but the internally encapsulated bgp messages cannot be parsed. It is evident that the captured bmp messages are very large, with individual BMP messages reaching 32kb in size.

**Comment by sbezverk** [comment]
@tankII  I was wondering if it would not too much to ask to run the same test but to use gobmp as a bmp collector with  `--v=9` . I would appreciate if you could do that and share generated log.

**Comment by tankII** [comment]
I  will  try  as  your advice  ,  but  where  is  gobmp  binary  release  ,  or  it's   source code  ? 
I found   code  in  https://github.com/osrg/gobmp.git    ,  only one file  and   depend package  github.com/osrg/gobgp/packet
not  available ,  can u  give  some advice , I am not very familiar with the go language .

**Comment by sbezverk** [comment]
@tankII  you can run it as a container, the latest posted on DockerHub. `docker.io/sbezverk/gobmp:latest` or you can clone the repo and build it `githib.com/sbezverk/gobmp`

**Comment by tankII** [comment]
gobmp built  from  githib.com/sbezverk/gobmp   v1.0.2  
gobgpd  version  is  3.17.0

first scenario  log :   keep  gobgpd  running  ,  then  restart  gobmp  
[gobgpd-reconnect.log](https://github.com/osrg/gobgp/files/12342784/gobgpd-reconnect.log)

second scenario  log :
keep  gobmp running   ,    delete bmp session  then add session again
[delete-bmpsession-add again.log](https://github.com/osrg/gobgp/files/12342785/delete-bmpsession-add.again.log)

**Comment by sbezverk** [comment]
@tankII thank you very much, collected logs will help me to better protect gobmp from such messages.

**Comment by sbezverk** [comment]
I decoded manually NLRI 71 message and I think it carries invalid length:
```
[ 
0x00, 0x04, type
0x00, 0x42, length 66       < -------------- Should be 60 bytes
0x02,  Protocol ID 1 byte
0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x5F, 0x50, Identifier 8 bytes

30 bytes
0x01, 0x00, TLV type 
0x00, 0x1A, TLV Length 26
0x02, 0x00, 0x00, 0x04, 0x00, 0x00, 0x5F, 0x50, 0x02, 0x01, 0x00, 0x04, 0xD3, 0x88, 0xBF, 0xFF, 0x02, 0x03, 0x00, 0x06, 0x21, 0x11, 0x36, 0x19, 0x12, 0x36,  Value

21 bytes
0x01, 0x09, Type
0x00, 0x11, Length 17
0x7F, 0x24, 0x09, 0x80, 0x1E, 0x00, 0xF0, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xCA 

] 
```

I think the next step is to see why gobgp sends  nlri 71 type 4 with 66 bytes instead of 60.

**Comment by tankII** [comment]
I did not study the gobgpd code very carefully, but provided a clue that if gobmp starts first and then gobgpd, there is no problem with bmp message parsing and all behavior is normal. The two previous scenarios where problems occurred were gobgpd sending the currently held BGP-LS information and constructing abnormal messages

**Comment by sbezverk** [comment]
Please give it a try one more time with this image: `docker.io/sbezverk/gobmp:debug_08-16-2023` , it should not panic anymore when a incorrect message NLRI 71 is sent.

**Comment by tankII** [comment]
yup  , gobmp  won't   panic  now  ,  when encounter  an incorrect message NLRI 71  will produce log  like below:

E0816 15:26:58.799459  417358 process-mp-update.go:145] failed to NLRI 71 with error: 64 bytes not enough to decode NLRI71 of 66 bytes

so  problem is still there :  
1 )    gobgpd   can't  aware  of  broken  bmp  session  and  try to  restablish  session  sometime 
2)     gobgpd  will  produce   incorrect message  with  NLRI 71   when   bmp session  recovery
